iocore/net/SSLCertLookup.cc

```diff
@@ -45,26 +47,19 @@
 #endif /* SSL_CTX_set_tlsext_ticket_key_cb */
 
 struct SSLAddressLookupKey {
-  explicit SSLAddressLookupKey(const IpEndpoint &ip) : sep(0)
+  explicit SSLAddressLookupKey(const IpEndpoint &ip)
   {
-    static const char hextab[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
-
-    int nbytes;
-    uint16_t port = ntohs(ip.port());
-
-    // For IP addresses, the cache key is the hex address with the port concatenated. This makes the lookup
-    // insensitive to address formatting and also allow the longest match semantic to produce different matches
-    // if there is a certificate on the port.
-    nbytes = ats_ip_to_hex(&ip.sa, key, sizeof(key));
-    if (port) {
-      sep           = nbytes;
-      key[nbytes++] = '.';
-      key[nbytes++] = hextab[(port >> 12) & 0x000F];
-      key[nbytes++] = hextab[(port >> 8) & 0x000F];
-      key[nbytes++] = hextab[(port >> 4) & 0x000F];
-      key[nbytes++] = hextab[(port)&0x000F];
+    // For IP addresses, the cache key is the hex address with the port concatenated. This makes the
+    // lookup insensitive to address formatting and also allow the longest match semantic to produce
+    // different matches if there is a certificate on the port.
+
+    ts::FixedBufferWriter w{key, sizeof(key)};
+    w.print("{}", ts::bwf::As_Hex(ip));
+    if (in_port_t port = ip.host_order_port(); port) {
+      sep = unsigned(w.size());
+      w.print(".{:x}", port);
     }
-    key[nbytes++] = 0;
+    w.print("\0");
```

Comment:
Just out of curiosity, would it be more efficient to allow something like

```
w.print('\0');
```

?

Reply:
Sigh. This was originally `w.write('\0');` for that reason, but maskit didn't like mixing the two. I don't think it's worth overloading `print` when `write` does exactly what you want.